google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -248,8 +248,23 @@ public Blob create(BlobInfo blobInfo, byte[] content, BlobTargetOption... option
   public Blob create(
       BlobInfo blobInfo, byte[] content, int offset, int length, BlobTargetOption... options) {
     Opts<ObjectTargetOpt> opts = Opts.unwrap(options).resolveFrom(blobInfo);
-    return internalDirectUpload(blobInfo, opts, ByteBuffer.wrap(content, offset, length))
-        .asBlob(this);
+    // Start the otel span to retain information of the origin of the request
+    OpenTelemetryTraceUtil.Span otelSpan =
+        openTelemetryTraceUtil.startSpan("create(BlobInfo, BlobTargetOption");
+    try (OpenTelemetryTraceUtil.Scope unused = otelSpan.makeCurrent()) {
+      return internalDirectUpload(
+              blobInfo,
+              opts,
+              ByteBuffer.wrap(content, offset, length),
+              openTelemetryTraceUtil.currentContext())
```

Comment:
What does currentContext() solve in this instrumentation?

Reply:
it is used to link this span to the child spans in internalDirectUpload

Reply:
since internalDirectUpload is used here and also in PCU I want the parent linked so we know where the original call came from.

Reply:
Gotcha; that makes sense. I don't think the automatic context propagation will work in the case of PCU

I was referring to: https://javadoc.io/doc/io.opentelemetry/opentelemetry-api/1.1.0/io/opentelemetry/api/trace/Tracer.html

> The automatic context propagation is done using Context which is a gRPC independent implementation for in-process Context propagation mechanism which can carry scoped-values across API boundaries and between threads. Users of the library must propagate the Context between different threads.

The last sentence is why you would need to do what you're doing now. Good call!